store/src/main/java/org/apache/rocketmq/store/timer/TimerMessageStore.java

```diff
@@ -503,7 +521,7 @@ private void checkBrokerRole() {
         BrokerRole currRole = storeConfig.getBrokerRole();
```

Comment:
I found that although lastBrokerRole is protected by volatile, storeConfig.getBrokerRole is not. Considering the replicas manager will control the store config directly, there may be a visibility issue. 

cc @GenerousMan